### PR TITLE
Allow for UDR model changes

### PR DESCRIPTION
**Pull request by eberseth**
1. Change default dead reckoning model from automotive to portable.
2. Add arguments on setup to specify model from application.

### CODE REVIEW

File: src/ubloxGPS.h [hunk 1]
```diff
@@ -613,7 +619,7 @@ class ubloxGPS
     // interrupt handler for tx ready signal from gps (if enabled)
     void txReadyHandler();
 
-    void on(void);
+    void on(ubx_dynamic_model_t model = UBX_DEFAULT_MODEL);
```
Comment (tstellanova) [inline src/ubloxGPS.h:655]: It would be good to start adding some header doc for important interface methods.

File: src/ubloxGPS.h [hunk 2]
```diff
@@ -643,8 +649,8 @@ class ubloxGPS
     uint32_t getTime(void);
     uint32_t getUTCTime();
     uint8_t  getSatellites(void);
-    int8_t   getGeoIdHeight(void);
-    uint8_t  getHDOP(void);
+    double   getGeoIdHeight(void);
```
Comment (tstellanova) [inline src/ubloxGPS.h:724]: What are the dimensional units for these methods (meters?) ? Would be good to add either to the method signature or to header doc.  Even better would be a dimensional type (something like meters_t or length_t, where the typedef is double)

File: src/ubloxGPS.h [hunk 3]
```diff
@@ -643,8 +649,8 @@ class ubloxGPS
...
+    double   getHDOP(void);
```
Comment (tstellanova) [inline src/ubloxGPS.h:731]: Would be good to at least add some description of the expected units and range.

File: src/ubloxGPS.h [hunk 4]
```diff
@@ -670,6 +676,8 @@ class ubloxGPS
     bool  getVersion(String& swVersion, String& hwVersion, String& extVersion);
     bool  setGNSS(uint8_t gnssMask);
     bool  setMode(ubx_dynamic_model_t dynModel);
+    bool  getMode(ubx_dynamic_model_t& dynModel);
+    bool  updateMode();
```
Comment (tstellanova) [inline src/ubloxGPS.h:787]: Just from the method signature, it's not clear to me what `updateMode()` does.  Does it flush changes made by `setMode()` ?  Header doc comments might help

**Review by tstellanova — CHANGES_REQUESTED** [review]

File: src/ubloxGPS.h (hunk 4 above)
Comment (eberseth) [reply src/ubloxGPS.h:787]: It is a method to tell the module to retrieve the data.  The driver is setup somewhat asynchronously so you'd make a command request and then filter the response from all of the mix of data sent from the module.

File: src/ubloxGPS.h (hunk 1 above)
Comment (eberseth) [reply src/ubloxGPS.h:655]: Added some doxygen comments

File: src/ubloxGPS.h (hunk 2 above)
Comment (eberseth) [reply src/ubloxGPS.h:724]: Meters.  Added some doxygen comments

File: src/ubloxGPS.h (hunk 3 above)
Comment (eberseth) [reply src/ubloxGPS.h:731]: Meters. Added some doxygen comments

**Review by tstellanova — APPROVED** [review]
Thank you for adding header docs!

**Pull request merged**